Jenkinsfile

```diff
@@ -377,6 +377,46 @@ pipeline {
                     }
                 }
 
+                stage('Hip Implicitgemm layouts Release') {
+                    agent{ label rocmnode("vega") }
+                    environment{
+                        cmd = '''
+                            # check out MLIR
+                            dir=$(pwd)
+                            cd
+                            git clone -b miopen-dialect https://github.com/ROCmSoftwarePlatform/llvm-project-mlir.git
```

Comment:
This should be built and installed in the dockerfile. Also, this should use a commit hash and not a branch name.

Reply:
Will building/installing inside the dockerfile cause it drag the build longer? I thought of the two alternatives a while ago and make it started in a jenkins file just so that non igemm stages do not suffer a performance penalty.

Besides, this review will not have everything in the final place at one go. Once the igemm got completely integrated, so that I can test not only NCHW layout, but also other layouts, I will move that to the dockerfile. Would this plan be okay?

Reply:
> Will building/installing inside the dockerfile cause it drag the build longer?

No, in fact it should be faster because the docker is cached on the nodes. Also, I assume its only temporary until this is upstreamed and provided by the llvmadgpu package.

> Besides, this review will not have everything in the final place at one go.

Yes, but this should be available for developers to run locally since this is being ran on jenkins.

Reply:
> No, in fact it should be faster because the docker is cached on the nodes. 
> Yes, but this should be available for developers to run locally since this is being ran on jenkins.

Sure, I can move this logic into dockerfile

> Also, I assume its only temporary until this is upstreamed and provided by the llvmadgpu package.

This will probably happen in a longer period. We upstream zero percent of the `miopen-dialect` inside of MLIR and this static library is a fat library that includes all the necessary llvm dependencies together. Desirably when we finished upstreaming, this library can ship together with rest of llvm libraries.